structs-in-rust/article.md

```diff
@@ -0,0 +1,443 @@
+
+#### Usage of Eq
+
+You may have noticed additional `Eq` and `PartialEq` traits.  Since we derived those traits, we can compare two `Point`s for equality:
+
+```
+let a = Point { x: 25, y: 50 };
+let b = Point { x: 100, y: 100 };
+let c = Point { x: 25, y: 50 };
+
+assert!(a == c);
+assert!(a != b);
+assert!(b != c);
+```
+
+#### Usage of Default
+
+For this example, we need to define another struct.
+
+```
+#[derive(Debug, PartialEq, Eq, Default)]
+struct DoubleBool(bool, bool);
+
+let point: Point = Default::default();
+let double_bool: DoubleBool = Default::default();
+
+assert_eq!(point, Point { x: 0, y: 0 });
+assert_eq!(double_bool, DoubleBool(false, false));
+```
+
+The most important lines in this snippet are those where we create `point` and `double_bool`. Because both `Point` and `DoubleBool` implement `Default`, we can invoke the `default()` method of the `Default` trait in each case to generate an empty value of type `Point` and `DoubleBool` respectively.
+
+*Exercise: remove explicit type annotations from lines 3 and 4 and see what the compiler tells you. Why so?*
+
+Traits deriving has its disadvantages.
+- Only a [small number of traits](https://doc.rust-lang.org/book/appendix-03-derivable-traits.html) is derivable. Though, [procedural macros](https://doc.rust-lang.org/stable/book/ch19-06-macros.html) allow for creation of custom `derive` attributes.
+- Sometimes the derived implementation doesn't match your expectations.
+
+That's why it is possible to implement a trait by yourself.
+
+### Implementing a trait
+
+When we tried to print a `Point` instance using `"{}"`, the compiler said that `Point` doesn't implement `std::fmt::Display`. This trait is similar to `std::fmt::Debug` but has a few differences:
+- It must be implemented manually.
+- It is expected to format values in a more pretty way, without containing any unnecessary information.
+
+You can think of the `Debug` and the `Display` as the formatters for programmers and users respectively.
+
+Let's add the manual implementation of the `Display`:
+
+```
+impl std::fmt::Display for Point {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "({}, {})", self.x, self.y)
+    }
+}
+
+let p = Point::new(0, 1);
+
+// format a pair
+assert_eq!(format!("{:?}", p), String::from("Point { x: 0, y: 0 }"));
+assert_eq!(format!("{}", p), String::from("(0, 0)"));
+```
+
+The syntax seems pretty natural. We `impl`ement the trait `std::fmt::Display` `for` the struct called `Point`. In the opened code block we then have to define every function that is declared in the trait. In our case, there are only one — `fmt`.
+
+## Struct methods
+
+Just like in Java or C++, in Rust, you can define **methods** for a particular type. This is done in an `impl` block.
+
+```
+impl Point {
+    // Associated function
+    fn new(x: i32, y: i32) -> Point {
+        Point { x, y }
+    }
+
+    // Immutable method
```

Comment:
I don't think terms "immutable method" and "mutable method" are common. Sounds kind of inaccurate, I dunno.
I wish there were succinct phrases for that, but nothing comes to mind.

If we are trying to avoid the term "borrow" for simplicity, the clunky alternative is "a method that can't modify struct instance" and "a method that can modify struct instance".

Reply:
Rephrased as suggested